clippy_lints/src/manual_assert.rs

```diff
@@ -62,25 +62,22 @@ impl<'tcx> LateLintPass<'tcx> for ManualAssert {
             };
             let cond_sugg = sugg::Sugg::hir_with_applicability(cx, cond, "..", &mut applicability).maybe_par();
             let semicolon = if is_parent_stmt(cx, expr.hir_id) { ";" } else { "" };
-            let sugg = format!("assert!({not}{cond_sugg}, {format_args_snip}){semicolon}");
-            // we show to the user the suggestion without the comments, but when applying the fix, include the
-            // comments in the block
+            let base_sugg = format!("assert!({not}{cond_sugg}, {format_args_snip}){semicolon}");
             span_lint_and_then(
                 cx,
                 MANUAL_ASSERT,
                 expr.span,
                 "only a `panic!` in `if`-then statement",
                 |diag| {
-                    // comments can be noisy, do not show them to the user
-                    if !comments.is_empty() {
-                        diag.tool_only_span_suggestion(
-                            expr.span.shrink_to_lo(),
-                            "add comments back",
-                            comments,
-                            applicability,
-                        );
-                    }
-                    diag.span_suggestion(expr.span, "try instead", sugg, applicability);
+                    // If we have comments to retain, include them in the final suggestion, if
+                    // not, don't.
+                    let suggestions = if comments.is_empty() {
+                        vec![(expr.span, base_sugg.clone())]
+                    } else {
+                        vec![(expr.span.shrink_to_lo(), comments), (expr.span, base_sugg.clone())]
```

Comment:
The current suggestion looks a little weird in the diagnostic with the colors
![image](https://github.com/user-attachments/assets/977e71fa-b705-4061-ba70-c8829d145b5f)

I suspect this is because the two replacements are at the same span, but I think it would be nice for this to also be properly indented.

We could probably combine these two replacement strings `comments + base_sugg` into one, and pass that combined string to the util function [`reindent_multiline`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/source/fn.reindent_multiline.html), with the indent obtained from the other util function [`indent_of(expr.span)`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/source/fn.indent_of.html).

From some local testing that looks like it should be enough
![image](https://github.com/user-attachments/assets/d29e446d-5e9a-4daf-ad7b-d011442b83b8)

Reply:
Hey @y21 good catch! i'm running on low capacity until after the break but i'll polish this up and ping you in the coming weeks 💪